core/src/main/java/org/elasticsearch/cluster/routing/allocation/AllocationService.java

```diff
@@ -85,7 +87,10 @@ public AllocationService(Settings settings, AllocationDeciders allocationDecider
         if (withReroute) {
             reroute(allocation);
         }
-        return new RoutingAllocation.Result(true, new RoutingTable.Builder().updateNodes(routingNodes).build().validateRaiseException(clusterState.metaData()));
+        RoutingAllocation.Result result = new RoutingAllocation.Result(true,
+                new RoutingTable.Builder().updateNodes(routingNodes).build().validateRaiseException(clusterState.metaData()));
+        logClusterHealthStateChange(clusterState, ClusterState.builder(clusterState).routingResult(result).build(), "applying started shards");
```

Comment:
maybe we can add a constructor to ClusterHealth which takes MetaData and RoutingTable so we won't have to build a cluster state here?